addons/misra.py

```diff
@@ -780,8 +780,14 @@ def misra_11_3(data):
         vt2 = token.astOperand1.valueType
         if not vt1 or not vt2:
             continue
-        if vt1.pointer == vt2.pointer and vt1.pointer > 0 and vt1.type != vt2.type and\
-           vt1.isIntegral() and vt2.isIntegral() and vt1.type != 'char':
+        if vt1.type == 'void' or vt2.type == 'void':
+            continue
+        if (vt1.pointer > 0 and vt1.type == 'record' and
+            vt2.pointer > 0 and vt2.type == 'record' and
+                vt1.typeScopeId != vt2.typeScopeId):
+            reportError(token, 11, 3)
+        if (vt1.pointer == vt2.pointer and vt1.pointer > 0 and
```

Comment:
it seems it would be a good idea to add a `elif` on this line to prevent duplicate warning messages.